lib/components/narrative/default/itinerary-summary.js

```diff
@@ -23,29 +40,48 @@ export default class ItinerarySummary extends Component {
       }
 
       // Add the mode icon
+      let title = leg.mode
+      if (leg.transitLeg) {
+        title = leg.routeShortName
+          ? `${leg.routeShortName}${leg.routeLongName ? ` - ${leg.routeLongName}` : ''}`
+          : leg.routeLongName
+      }
+      const style = {
+        margin: '0px',
+        padding: '3px',
+        height: '24px',
+        width: '24px',
+        backgroundColor: getModeColor(leg.mode)
+      }
+      if (i === 0) {
+        style.borderTopLeftRadius = '4px'
+        style.borderBottomLeftRadius = '4px'
+      }
+      if (i === itinerary.legs.length - 1) {
+        style.borderTopRightRadius = '4px'
+        style.borderBottomRightRadius = '4px'
+      }
       blocks.push(
-        <div key={blocks.length} className='summary-block mode-block'>
+        <div
+          style={style}
+          title={title}
+          key={blocks.length}
+          className='summary-block mode-block'
+        >
           <LegIcon leg={leg} />
         </div>
       )
 
-      // If a transit leg, add the name (preferably short; long if needed)
-      if (leg.transitLeg) {
+      // If a transit leg, add the name (only if short name exists)
+      if (leg.transitLeg && leg.routeShortName) {
```

Comment:
This should not repeat a non-transit mode IMO:
 
![image](https://user-images.githubusercontent.com/56846598/90543932-5fdc5880-e154-11ea-9082-6d5150ede694.png)

Query: http://localhost:9966/#/?ui_activeSearch=gzw0a0kme&ui_activeItinerary=0&fromPlace=1222%20SW%20Davenport%20St%2C%20Portland%3A%3A45.505143833370234%2C-122.69187927246095&toPlace=3504%20NE%20Us%20Grant%20Pl%2C%20Portland%3A%3A45.53797835175697%2C-122.62819290161134&date=2020-08-18&time=11%3A35&arriveBy=false&mode=WALK%2CBUS%2CTRAM%2CRAIL%2CGONDOLA&showIntermediateStops=true&maxWalkDistance=1207&optimize=QUICK&walkSpeed=1.34&ignoreRealtimeUpdates=true&companies=&bannedRoutes=&numItineraries=3&intermediatePlaces=Rose%20Quarter%20TC%20MAX%20Station%2C%20Portland%3A%3A45.530147%2C-122.665225&otherThanPreferredRoutesPenalty=900&preferredRoutes=

Reply:
This trip is for an intermediate place, so there are two distinct legs in the itinerary for walking. I somewhat disagree, but in general I think this trip summary component is not quite finished and needs refinement, which I think we should handle separately.

Reply:
Ooh ok. In that case the summary band would need to show an intermediate place indicator/separator.